### PR TITLE
refactor(localized-multiline-text-input): to use hooks

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Refactors the `LocalizedMultilineTextInput` to use hooks.

### CODE REVIEW

File: src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js [hunk 1]
```diff
@@ -27,247 +28,228 @@ import RequiredValueErrorMessage from './required-value-error-message';
...
+    getHasErrorOnRemainingLanguages(props.errors, props.selectedLanguage);
+  const hasWarningInRemainingLanguages =
+    props.hasWarning ||
+    getHasWarningOnRemainingLanguages(props.warnings, props.selectedLanguage);
+
+  if (hasErrorInRemainingLanguages || hasWarningInRemainingLanguages) {
+    if (!areLanguagesOpened) {
+      // this update within render replaces the old `getDerivedStateFromProps` functionality
+      // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
+      toggleLanguages();
```
Comment (montezume) [inline src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js]: I find this waaaaaaaaaaaaaaaaay easier to read than the old `getDerivedStateFromProps`

File: src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js [hunk 2]
```diff
@@ -27,247 +28,228 @@ import RequiredValueErrorMessage from './required-value-error-message';
...
-
-  static isEmpty = isEmpty;
-
-  static omitEmptyTranslations = omitEmptyTranslations;
-
-  static isTouched = isTouched;
+    {}
+  );
+
+  const [expandedTranslations, setExpandedTranslations] = React.useState(
```
Comment (emmenko) [inline src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js]: We should use `useReducer`

File: src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js [hunk 3]
```diff
@@ -27,247 +28,228 @@ import RequiredValueErrorMessage from './required-value-error-message';
...
+    // useToggleState's default is `true`, but we want `false`
+    false;
+
+  const [areLanguagesOpened, toggleLanguages] = useToggleState(
+    defaultExpansionState
+  );
+
+  const toggleLanguage = React.useCallback(
+    language => {
+      const newExpandedTranslations = {
```
Comment (emmenko) [inline src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js]: This will be done within the reducer

**Comment by montezume** [comment]
Updates to use reducer

File: src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js [hunk 4]
```diff
@@ -21,253 +22,244 @@ import LanguagesControl from './languages-control';
...
+    false;
 
-  static getName = getName;
+  const [areLanguagesOpened, toggleLanguages] = useToggleState(
+    defaultExpansionState
+  );
 
-  static defaultProps = {
-    horizontalConstraint: 'scale',
+  const toggleLanguage = language => {
```
Comment (emmenko) [inline src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js]: Nit: memoize it with `useCallback`

File: src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js [hunk 5]
```diff
@@ -21,253 +22,244 @@ import LanguagesControl from './languages-control';
...
+      const newState = Object.keys(state).reduce((translations, locale) => {
+        return {
+          [locale]: true,
+          ...translations,
+        };
+      }, {});
+      return newState;
+    }
+    default:
+      throw new Error();
```
Comment (emmenko) [inline src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js]:
```suggestion
      return state
```

File: src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js [hunk 6]
```diff
@@ -21,253 +22,244 @@ import LanguagesControl from './languages-control';
 import TranslationInput from './translation-input';
 import RequiredValueErrorMessage from './required-value-error-message';
 
+const expandedTranslationsReducer = (state, action) => {
```
Comment (emmenko) [inline src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js]: Nice 👌

File: src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js [hunk 7]
```diff
@@ -21,253 +22,244 @@ import LanguagesControl from './languages-control';
...
 
-  static getName = getName;
+  const [areLanguagesOpened, toggleLanguages] = useToggleState(
+    defaultExpansionState
+  );
 
-  static defaultProps = {
-    horizontalConstraint: 'scale',
+  const toggleLanguage = language => {
+    expandedTranslationsDispatch({ type: 'toggle', language });
```
Comment (emmenko) [inline src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js]: Nit: if we model the actions a-la redux, we should use `{ type, payload }`

```suggestion
    expandedTranslationsDispatch({ type: 'toggle', payload: language });
```

**Pull request merged**